### PR TITLE
fix: [PIE-4273]: Run pipeline context added and service entity as runtime implemented

### PR DESCRIPTION
**Pull request by bhavya-sinha**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/70-pipeline/context/RunPipelineFormContext.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { noop } from 'lodash-es'
+import type { PipelineInfoConfig } from 'services/cd-ng'
+
+export interface RunPipelineFormContextValues<T = unknown> {
+  template: PipelineInfoConfig
+  updateTemplate: (data: T, path: string) => void
```
Comment (vkbansal) [inline src/modules/70-pipeline/context/RunPipelineFormContext.tsx:14]:
```suggestion
  updateTemplate<T = unknown>(data: T, path: string): void
```

File: src/modules/70-pipeline/context/RunPipelineFormContext.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { noop } from 'lodash-es'
+import type { PipelineInfoConfig } from 'services/cd-ng'
+
+export interface RunPipelineFormContextValues<T = unknown> {
```
Comment (vkbansal) [inline src/modules/70-pipeline/context/RunPipelineFormContext.tsx:12]:
```suggestion
export interface RunPipelineFormContextValues {
```

File: src/modules/70-pipeline/context/RunPipelineFormContext.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  template: PipelineInfoConfig
+  updateTemplate: (data: T, path: string) => void
+}
+
+export const RunPipelineFormContext = React.createContext<RunPipelineFormContextValues>({
+  template: {} as PipelineInfoConfig,
+  updateTemplate: () => noop
```
Comment (vkbansal) [inline src/modules/70-pipeline/context/RunPipelineFormContext.tsx:19]:
```suggestion
  updateTemplate: noop
```

File: src/modules/70-pipeline/components/PipelineStudio/__tests__/StepUtils.test.ts [hunk 4]
```diff
@@ -10,7 +10,8 @@ import has from 'lodash-es/has'
 import { get } from 'lodash-es'
 import { StepViewType } from '@pipeline/components/AbstractSteps/Step'
 import type { PipelineInfoConfig } from 'services/cd-ng'
-import { validateCICodebase, getErrorsList, validatePipeline, clearRuntimeInput } from '../StepUtil'
+import { clearRuntimeInput } from '@pipeline/utils/runPipelineUtils'
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/PipelineStudio/__tests__/StepUtils.test.ts:13]: Please move the related test cases too

Comment (bhavya-sinha) [reply src/modules/70-pipeline/components/PipelineStudio/__tests__/StepUtils.test.ts]: done

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**